src/main/java/frc/robot/leds/LedStrip.java

```diff
@@ -0,0 +1,156 @@
+     * It will be used when in the solid mode and as the
+     * initial color in the blink and fade modes
+     * @param color
+     */
+    private void setPrimary(Color color) {
+        primary = color;
+    }
+
+    /**
+     * Sets the secondary color of the LEDs.
+     * It will be used as the second color in the blink and fade modes
+     * @param color
+     */
+    private void setSecondary(Color color) {
+        secondary = color;
+    }
+
+    private void setMode(LedMode mode) {
+        this.mode = mode;
+    }
+
+    /**
+     * Sets the led blink time.
+     * @param blinkTime Blink time in seconds
+     */
+    private void setBlinkTime(double blinkTime) {
+        this.blinkTime = blinkTime;
+    }
+
+    /**
+     * Sets the duration of the fade effect.
+     * @param duration Fade duration in seconds.
+     */
+    private void setFadeDuration(double duration) {
+        fadeDuration = duration;
+    }
+
+    private void setSolidColor(Color color) {
+        var appliedColor = new edu.wpi.first.wpilibj.util.Color(color.getRed(), color.getGreen(), color.getBlue());
+        for (int i = 0; i < stripLength; i++) {
+            ledBuffer.setLED(i, appliedColor);
+        }
+        ledStrip.setData(ledBuffer);
+    }
+
+    /**
+     * Updates fadeColor to the correct color for a fade effect
+     * at the current time by interpolating the two given colors.
+     * @param initial Initial color.
+     * @param goal    Final Color.
+     */
+    private void colorInterpolation(Color initial, Color goal) {
+        var initialHSB = Color.RGBtoHSB(initial.getRed(), initial.getGreen(), initial.getBlue(), new float[3]);
+        var goalHSB = Color.RGBtoHSB(goal.getRed(), goal.getGreen(), goal.getBlue(), new float[3]);
+
+        Translation3d initialPoint = new Translation3d(initialHSB[0], initialHSB[1], initialHSB[2]);
+        Translation3d goalPoint = new Translation3d(goalHSB[0], goalHSB[1], goalHSB[2]);
+
+        double d = initialPoint.getDistance(goalPoint) / fadeDuration;
+        double t = d / (initialPoint.minus(goalPoint)).getNorm();
+
+        Translation3d solution = initialPoint.interpolate(goalPoint, t);
+        fadeColor = new Color(
+                (int)solution.getX(),
+                (int)solution.getY(),
+                (int)solution.getZ()
+        );
+    }
+
+    private void setRainbow() {
+        int rainbowHue = 0;
+        for (int i = 0; i < stripLength; i++) {
+            ledBuffer.setHSV(i, rainbowHue, 255, 180);
+            rainbowHue += (180 / stripLength);
+            ledStrip.setData(ledBuffer);
+            rainbowHue %= 180;
+        }
+    }
+
+    @Override
+    public void periodic() {
+        switch (mode){
+            case SOLID:
+                setSolidColor(currentColor);
+
+            case BLINK:
+                if (currentColor == primary) currentColor = secondary;
+                else currentColor = primary;
+
+                timer.advanceIfElapsed(blinkTime);
```

Comment:
This doesn't stop the thread, it returns a boolean that says whether the time has elapsed.